MyLibrary/BusinessLogicTests/AuthorServiceTests.cs

```diff
@@ -0,0 +1,209 @@
+
+            // Act
+            uint? actualId = libraryService.Remove(id);
+            Author actualAuthor = libraryService.GetById(id);
+
+            // Assert
+            Assert.AreEqual(expectedId, actualId);
+            Assert.IsNull(actualAuthor);
+        }
+
+        /// <summary>
+        /// Tests the method RemoveAuthor
+        /// Incorrect input value for author id
+        /// Returned null
+        /// </summary>
+        [TestMethod]
+        public void RemoveAuthor_NotExistedAuthorId_ReturnedNull()
+        {
+            // Arrange
+            uint authorIdForDelete = 100;
+            Author author = new Author("TestBook");
+            var mockDataProvider = new Mock<IDataProvider>();
+            mockDataProvider.SetupGet(x => x.GetAuthors).Returns(new List<Author> { author });
+            var libraryService = new AuthorService(mockDataProvider.Object);
+
+            // Act
+            uint? actualId = libraryService.Remove(authorIdForDelete);
+
+            // Assert
+            Assert.IsNull(actualId);
+        }
+
+        /// <summary>
+        /// Tests the method AddAuthor
+        /// Correct input values
+        /// Returned correct author
+        /// </summary>
+        /// <param name="name">The name of author</param>
+        [TestMethod]
+        [DataRow("Test Author")]
+        public void AddAuthor_CorrectInput_ReturnedCorrectAuthor(string name)
+        {
+            // Arrange
+            Author authorForAdding = new Author(name);
+            Author author = new Author("TestAuthor2");
+            var mockDataProvider = new Mock<IDataProvider>();
+            mockDataProvider.SetupGet(x => x.GetAuthors).Returns(new List<Author> { author });
+            var libraryService = new AuthorService(mockDataProvider.Object);
+            int countOfAuthorsBeforeAdding = libraryService.GetAll().ToList<Author>().Count;
+
+            // Act
+            uint idOfAddedAuthor = libraryService.Add(authorForAdding);
+            int countOfAuthorsAfterAdding = libraryService.GetAll().ToList<Author>().Count;
+            Author addedAuthor
+                = libraryService.GetAll().FirstOrDefault(someAuthor => someAuthor.Id == idOfAddedAuthor);
+
+            // Assert
+            Assert.AreEqual(countOfAuthorsBeforeAdding + 1, countOfAuthorsAfterAdding);
+            Assert.AreEqual(authorForAdding, addedAuthor);
+        }
+
+        /// <summary>
+        /// Tests the method UpdateAuthor
+        /// Correct input values
+        /// Returned correct author
+        /// </summary>
+        /// <param name="authorIdForUpdating">The identifier of author for updating</param>
+        /// <param name="name">The name of author</param>
+        [TestMethod]
+        [DataRow(1u, "TestAuthor")]
+        public void UpdateAuthor_CorrectInput_ReturnedCorrectAuthor(uint authorIdForUpdating, string name)
+        {
+            // Arrange
+            Author author = new Author("TestAuthorUpdate");
```

Comment:
in such case use, var in the left part of expression as the type is clear from the right side, it will make your life slightly easier.